blaze/expr/collections.py

```diff
@@ -424,10 +424,36 @@ def join(lhs, rhs, on_left=None, on_right=None, how='inner'):
 
 join.__doc__ = Join.__doc__
 
+class IsIn(Expr):
+    """
+    """
+    __slots__ = '_hash', '_child', '_key'
+
+    def __init__(self, child, key):
+        super(IsIn, self).__init__(child, frozenset(key))
+
+    @property
+    def dshape(self):
+        """ self._child.dshape * number of matched results
+        """
+        return DataShape(*(self._child.shape + (datashape.bool_,)))
+
+    @property
+    def _len(self):
+        """ number of matched results """
+        pass
+
+    def __str__(self):
+        return "%s.isin(%s)" % (self._child, self._key)
```

Comment:
small nitpick:

it might be nice to repr `self` like this:

```
'%s.isin({%s})' % (self._child, ', '.join(map(repr, self._key)))
```
so that `_key` always looks like a `set`.